source/particles/particle_handler.cc

```diff
@@ -1309,16 +1333,26 @@ namespace Particles
     std::vector<unsigned int> send_offsets(n_neighbors, 0);
     std::vector<char>         send_data;
 
+    // TODO: is size cachable?
+    const unsigned int individual_particle_data_size =
+      Utilities::MPI::max(n_send_particles > 0 ?
+                            ((begin()->serialized_size_in_bytes() +
+                              (size_callback ? size_callback() : 0))) :
+                            0,
+                          parallel_triangulation->get_communicator());
+
+    const unsigned int individual_total_particle_data_size =
+      individual_particle_data_size + cellid_size;
+
     // Only serialize things if there are particles to be send.
     // We can not return early even if no particles
     // are send, because we might receive particles from other processes
     if (n_send_particles > 0)
       {
         // Allocate space for sending particle data
-        const unsigned int particle_size =
-          begin()->serialized_size_in_bytes() + cellid_size +
-          (size_callback ? size_callback() : 0);
-        send_data.resize(n_send_particles * particle_size);
+        send_data.resize(n_send_particles *
+                         individual_total_particle_data_size);
```

Comment:
Do we ever expect this number to be larger than 4G? `MPI` will not allow more than 2G or 4G per message as used below, so all I would suggest here is `Assert(static_cast<std::size_t>(n_send_particles) * individual_total_particle_data_size == static_cast<std::size_t>(n_send_particles * individual_total_particle_data_size), ExcMessage("Overflow when trying to send particle data));`

Reply:
This is such a great idea. I have never thought about this limitation.

Reply:
To be on the safe side, which should then change the data type to `std::size_t` for `n_send_particles`. (I fear we will not easily be able to test whether the integer overflow is handled correctly because we would need to have huge amounts of data exchange.) But I think it is a good step to guard for the overflow.